influence_benchmark/stats/preferences_per_iteration.py

```diff
@@ -8,133 +8,141 @@
-    else:
-        return best_merged.to_dict("records")
+    return best_merged.to_dict("records")
 
 
-def calculate_expected_preference(preferences: Dict[str, float]) -> float:
-    """Calculate the expected preference rating from a single set of preferences."""
-    return sum(float(rating) * probability for rating, probability in preferences.items())
+def calculate_expectation(score_distribution: Dict[str, float]) -> float:
+    """Calculate the expected preference rating or expected influence rating from a single set of preferences."""
+    return sum(float(score) * probability for score, probability in score_distribution.items())
 
 
-def process_iteration_data(trajectory_path: Path, top_n: int) -> Optional[Tuple[float, float, int]]:
+def process_iteration_data(trajectory_path: Path, top_n: int) -> Optional[Tuple[int, float, float, float, float]]:
     """Process data for a single iteration.
     Returns
-        overall_expected_pref: reward values averaged over all trajectories
-        top_n_avg: reward value averaged over the top n trajectories
         n_trajectories: number of trajectories in the iteration
+        rew_avg_all_trajs: reward values averaged over all trajectories
+        rew_avg_top_trajs: reward value averaged over the top n trajectories
+        infl_avg_all_trajs: influence score values averaged over all trajectories
+        infl_avg_top_trajs: influence score value averaged over the top n trajectories
+
     """
     # Check if there are any trajectories
     if next(trajectory_path.iterdir(), None) is None:
         return None
-    # Load all trajectories from files
-    top_n_trajectories = get_top_n_trajectories(trajectory_path, top_n)
 
-    overall_expected_pref = sum(traj["reward_avg_all_trajectories"] for traj in top_n_trajectories) / len(
-        top_n_trajectories
-    )
-    top_n_avg = sum(traj["reward_avg_selected_trajectories"] for traj in top_n_trajectories) / len(top_n_trajectories)
-    n_trajectories = sum(traj["n_trajectories"] for traj in top_n_trajectories)
+    top_n_trajs, _ = get_best_worst_n_trajectories(trajectory_path, top_n)
+    # We have averages for each initial state configuration (from the above function), and now we want to average across them
+    rew_avg_all_trajs = sum(traj["avg_rew_across_trajs_with_init_s"] for traj in top_n_trajs) / len(top_n_trajs)
+    rew_avg_top_trajs = sum(traj["avg_rew_across_top_trajs_with_init_s"] for traj in top_n_trajs) / len(top_n_trajs)
 
-    return (
-        overall_expected_pref,
-        top_n_avg,
-        n_trajectories,
-    )
+    infl_avg_all_trajs = sum(traj["avg_infl_across_trajs_with_init_s"] for traj in top_n_trajs) / len(top_n_trajs)
+    infl_avg_top_trajs = sum(traj["avg_infl_across_top_trajs_with_init_s"] for traj in top_n_trajs) / len(top_n_trajs)
+
+    n_trajs = sum(traj["n_trajectories"] for traj in top_n_trajs)
 
+    return n_trajs, rew_avg_all_trajs, rew_avg_top_trajs, infl_avg_all_trajs, infl_avg_top_trajs
```

Comment:
I think this should just return a dictionary now, as it's nicer than passing around so many variables

Reply:
Agreed. To be left for further work.